File: frontend/src/types/accounting.ts
export interface AccountingProviderInfo {
  key: string;
  name: string;
  description: string;
  supports_sync: boolean;
}

export interface AccountingStatus {
  provider: string;
  connected: boolean;
  last_sync_at: string | null;
  error: string | null;
  details: Record<string, unknown>;
}

export interface SyncResult {
  success: boolean;
  records_synced: number;
  records_failed: number;
  sync_log_id: string | null;
  error_message: string | null;
  details: Record<string, unknown> | null;
}

export interface ProviderAccount {
  id: string;
  name: string;
  account_type: string;
  number: string | null;
  is_active: boolean;
}

export interface AccountMapping {
  internal_id: string;
  internal_name: string;
  provider_id: string | null;
  provider_name: string | null;
}

export interface QBOConnectResponse {
  authorization_url: string;
  state: string;
}

File: backend/app/schemas/accounting.py
from datetime import datetime

from pydantic import BaseModel


class AccountingProviderInfo(BaseModel):
    key: str
    name: str
    description: str
    supports_sync: bool


class AccountingConfigResponse(BaseModel):
    provider: str
    connected: bool
    last_sync_at: datetime | None = None
    error: str | None = None
    details: dict = {}


class AccountingProviderUpdate(BaseModel):
    provider: str  # none, sage_csv, quickbooks_online


class SyncRequest(BaseModel):
    sync_type: str  # customers, invoices, payments, bills, bill_payments, inventory
    direction: str = "push"  # push, pull, bidirectional
    date_from: datetime | None = None
    date_to: datetime | None = None


class SyncResultResponse(BaseModel):
    success: bool
    records_synced: int = 0
    records_failed: int = 0
    sync_log_id: str | None = None
    error_message: str | None = None
    details: dict | None = None


class ProviderAccountResponse(BaseModel):
    id: str
    name: str
    account_type: str
    number: str | None = None
    is_active: bool = True


class AccountMappingResponse(BaseModel):
    internal_id: str
    internal_name: str
    provider_id: str | None = None
    provider_name: str | None = None


class AccountMappingUpdate(BaseModel):
    internal_id: str
    provider_id: str


class QBOConnectResponse(BaseModel):
    authorization_url: str
    state: str
